packages/table/src/Row.tsx

```diff
@@ -0,0 +1,323 @@
+            hasSeenRow = true;
+          }
+        }
+      });
+
+      if (hasSeenRow) {
+        renderedChildren[0] = React.cloneElement(renderedChildren[0], {
+          children: (
+            <>
+              {chevronButton}
+              <span className={truncation}>
+                {renderedChildren[0].props.children}
+              </span>
+            </>
+          ),
+          className: cx(displayFlex, className),
+          key: `${indexRef.current}-${renderedChildren[0].props.children}`,
+        });
+      }
+
+      return renderedChildren;
+    }, [children, disabled, className, isExpanded, setIsExpanded]);
+
+    // Iterating over children twice because generated memoized values have different dependants
+    const nestedRows = React.useMemo(() => {
+      let hasSeenFirstCell = false;
+      const nestedRows: Array<React.ReactElement> = [];
+
+      React.Children.forEach(children, (child, index) => {
+        if (isComponentType(child, 'Cell') && !hasSeenFirstCell) {
+          hasSeenFirstCell = true;
+        }
+
+        if (child != null && isComponentType(child, 'Row')) {
+          nestedRows.push(
+            React.cloneElement(child, {
+              ref: nodeRef,
+              isAnyAncestorCollapsed: isAnyAncestorCollapsedProp || !isExpanded,
+              indentLevel: indentLevel + 1,
+              key: `${indexRef.current}-${indentLevel}-${index}`,
+            }),
+          );
+        }
+      });
+
+      return nestedRows;
+    }, [isAnyAncestorCollapsedProp, isExpanded, indentLevel, children]);
+
+    // Depending on network speed, will noticeably render columns with incorrect
+    // alignment, would rather wait for proper information before rendering
+    if (!columnInfo) {
+      return null;
+    }
+
+    const shouldAltRowColor = data && data.length >= 10 && !hasNestedRows;
+
+    const alignmentStyles = Object.entries(
+      columnInfo,
+    ).map(([key, { dataType }]) => styleColumn(key, dataType!));
+
+    const rowClassName = cx(
+      rowStyle,
+      getIndentLevelStyle(indentLevel),
+      [...alignmentStyles],
+      {
+        [altColor]: shouldAltRowColor,
+        [disabledStyle]: disabled,
+      },
+      className,
+    );
+
+    const ariaExpanded =
+      nestedRows.length > 0
+        ? {
+            ['aria-expanded']: isExpanded,
+          }
+        : undefined;
+
+    return (
+      <>
+        <tr
+          className={rowClassName}
+          aria-disabled={disabled}
+          ref={ref}
+          key={indexRef.current}
+          {...ariaExpanded}
+          {...rest}
+        >
+          {renderedChildren}
+        </tr>
+        {nestedRows && nestedRows.length > 0 && (
+          <Transition
+            in={isExpanded && !isAnyAncestorCollapsedProp}
+            timeout={150}
+            nodeRef={nodeRef}
+          >
+            {(state: string) => {
+              return (
+                <>
+                  {nestedRows?.map(element =>
```

Comment:
Should be redundant due to the conditional it's rendered within?

Reply:
👍